### PR TITLE
Added ability to reset timestamp value to null using value prop

### LINKED ISSUE #2175: Not able to reset timestamp/value to null for a controlled DateTimePicker component
https://yedhin.neetorecord.com/watch/7c12e97c-b829-4ca9-ab9b-9d7cef8d8e2f

@josephmathew900 _a Please assign this to someone.

cc: @AbhayVAshokan

### PR DESCRIPTION
**Pull request by gaagul**
- Fixes #2175 

**Description**

- Added: ability to reset timestamp value to null using value prop

**Checklist**
~~- [ ] I have made corresponding changes to the documentation.~~
~~- [ ] I have updated the types definition of modified exports.~~
- [x] I have verified the functionality in some of the neeto web-apps.
~~- [ ] I have added tests that prove my fix is effective or that my feature works.~~
~~- [ ] I have added proper `data-cy` and `data-testid` attributes.~~
- [x] I have added the necessary label (`patch`/`minor`/`major` - If package publish
      is required).

**Reviewers**

@josephmathew900 _a

### CODE REVIEW

**Comment by gaagul** [comment]
@gaagul _a

**Comment by gaagul** [comment]
@josephmathew900 _a Please review.

File: src/components/TimePickerInput/index.jsx [hunk 1]
```diff
@@ -55,14 +56,17 @@ const TimePickerInput = forwardRef(
     const errorId = `error_${id}`;
 
     useEffect(() => {
-      if (isNotPresent(inputValue) && isNotPresent(defaultValue)) return;
+      const value = inputValue || defaultValue;
+      if (isNil(value) || isNotPresent(value)) {
```
[the remark this answers is not in the document]
Comment (gaagul) [reply src/components/TimePickerInput/index.jsx]: Demo: https://gaagul-gigi.neetorecord.com/watch/aa5ababe-9614-4848-9f7c-da8a71c491c3?time=12

**Review by josephmathew900 — APPROVED** [review]
LGTM.

**Pull request merged**